### ISSUE TITLE
`prefer-const` false positive when reassigned in `switch`

### ISSUE THREAD

**Issue by Skalman** [denoland/deno_lint#665]
`prefer-const` emits an error, even though the variable is assigned in the `case`.

```js
let a; // 'a' is never reassigned
switch (Date.now() % 2) {
  case 0:
    a = 1;
}

console.log(a);
```

Changing the `switch` to an `if`works as expected.

**Comment by magurotuna** [comment]
Thanks for reporting that too. I'll fix it.
